variant/effect.go

```diff
@@ -0,0 +1,276 @@
+package variant
+
+import (
+	"github.com/vertgenlab/gonomics/dna"
+)
+
+// CodingChange describes a change of a protein sequence caused by
+// a mutation affecting the coding sequence of the corresponding gene.
+// CodingPos and ProteinPos refer to the zero-based start position of
+// the change. The CodingPos always refers to the position changed in
+// origin DNA sequence, however ProteinPos is not always equal to
+// CodingPos / 3. If a change would result in a RemovedAa[0] == AddedAa[0]
```

Comment:
If you have the following deletion

```
Prot Pos:  0  1  2  3-n
Sequence: ATGATGGGG~~~~~
Deletion:      ***
New Seq:  ATGATG~~~~~~~
```

Since the deletion did not change the first codon, the protein pos is reported as the first codon that was deleted. In this case, it would be the 'GGG' even though the DNA deletion actually removed part of the 'ATG'.